packages/cds-odata-annotation-converter/src/transforms/annotation/flattened.ts

```diff
@@ -46,18 +47,29 @@ export function convertFlattenedPath(
     //             Property    Property      Enum Member
     let root: Element | undefined;
     let parent: Element | undefined;
-    const expandedStructures = convertToExpandedStructure(state, segments, value?.range);
+    const expandedStructures = convertToExpandedStructure(state, segments, value);
     for (const expandedStructure of expandedStructures) {
         if (parent) {
-            const record: Element = createElementNode({
-                name: Edm.Record,
-                range: expandedStructure.element.range ?? undefined,
-                content: [expandedStructure.element],
-                contentRange: expandedStructure.element.range ?? undefined
-            });
-            // property content range should include only child range
-            parent.contentRange = record.range ?? undefined;
-            parent.content.push(record);
+            if (expandedStructure.kind === 'record-type') {
+                if (expandedStructure.element.attributes['Type'].nameRange?.start && parent.contentRange?.end) {
+                    const range: Range = {
+                        start: expandedStructure.element.attributes['Type'].nameRange?.start,
+                        end: parent.contentRange?.end
+                    };
+                    parent.contentRange = copyRange(range);
+                }
```

Comment:
We already calculate the correct range when creating record, we could just reuse that.

```suggestion
                if (expandedStructure.element.range) {
                    parent.contentRange = copyRange(expandedStructure.element.range);
                }
```

Reply:
adapted